ingest/subsample.py

```diff
@@ -0,0 +1,120 @@
+import copy
+import json
+import pprint
+import random
+import re
+
+import numpy as np
+import pandas
+from clusters import Clusters
+from ingest_files import IngestFiles
+
+
+class SubSample(IngestFiles):
+    ALLOWED_FILE_TYPES = ['text/csv',
+                          'text/plain', 'text/tab-separated-values']
+    MAX_THRESHOLD = 100_000
+    SUBSAMPLE_THRESHOLDS = [MAX_THRESHOLD, 10_000, 1_000]
+
+    def __init__(self, *, cluster_file=None, cell_metadata_file=None):
+        IngestFiles.__init__(self,
+                             cluster_file, self.ALLOWED_FILE_TYPES, open_as='pandas')
+        self.dermine_coordinates_and_cell_names()
+        self.cell_metadata_file = cell_metadata_file
+        self.correct_annot_types()
+
+    def correct_annot_types(self):
+        """Ensures that numeric columns are rounded to 2 decimals points and
+        group annotations are strings
+        """
+        numeric_columns = self.file.xs("numeric", axis=1, level=1,
+                                       drop_level=False).columns.tolist()
+        self.file[numeric_columns] = self.file[numeric_columns].round(2)
+        group_columns = self.file.xs("group", axis=1, level=1,
+                                     drop_level=False).columns.tolist()
+        self.file[group_columns] = self.file[group_columns].astype(str)
+
+    def prepare_cell_metadata(self):
+        """ Does an inner join on cell and cluster file """
+        if self.cell_metadata_file is not None:
+            self.merge_df(self.cell_metadata_file,
+                          self.file[self.coordinates_and_cell_names])
+
+    def dermine_coordinates_and_cell_names(self):
```

Comment:
```suggestion
    def determine_coordinates_and_cell_names(self):
```

And similarly for other references to this method.